buchschloss/config/validation.py

```diff
@@ -40,10 +42,45 @@ def is_optionlist(value, *options):
     return val
 
 
-def is_task_list(value, _tasks=('backup', 'ftp_backup', 'late_books', 'http_backup')):
-    """check whether the value is a list of tasks"""
-    # TODO: make ``_tasks`` a mapping directly to the functions
-    return validator.check('optionlist{}'.format(_tasks), value)
+def is_script_spec(value,
+                   with_time=False,
+                   single=False,
+                   suffixes=('lua', 'py'),
+                   default_suffix='lua'):
+    """check whether the value is syntactically a script spec and return components"""
+    spec_regex_parts = [
+        r'^\s*(?P<name>[-\w ]+)(?::(?P<function>\w+))?',
+        '|'.join(suffixes).join((r'(?:!(?P<type>', '))?')),
+    ]
+    if with_time:
+        spec_regex_parts.append(r'@(?P<invocation>\d+(?::\d+(?::\d+)?)?)')
+    spec_regex = re.compile(''.join(spec_regex_parts) + r'\s*$')
+    if isinstance(value, str):
+        value = value.split(',')
```

Comment:
AFAIK, if it contains commas, it'll be converted to a list by configobj. Just ``= [value]`` here.